lib/internal/fs/watch/linux.js

```diff
@@ -0,0 +1,244 @@
+'use strict';
+
+const { SafeMap, Symbol, StringPrototypeStartsWith, SymbolAsyncIterator, Promise } = primordials;
+
+const {
+  ERR_FEATURE_UNAVAILABLE_ON_PLATFORM,
+  codes: { ERR_INVALID_ARG_TYPE, ERR_INVALID_ARG_VALUE },
+} = require('internal/errors');
+const { kEmptyObject } = require('internal/util');
+const { validateString, validateObject, validateBoolean, validateAbortSignal } = require('internal/validators');
+const { EventEmitter } = require('events');
+const path = require('path');
+
+const kFSWatchStart = Symbol('kFSWatchStart');
+
+let internalSync;
+let internalPromises;
+
+function lazyLoadFsPromises() {
+  internalPromises ??= require('fs/promises');
+  return internalPromises;
+}
+
+function lazyLoadFsSync() {
+  internalSync ??= require('fs');
+  return internalSync;
+}
+
+async function traverse(dir, files = new SafeMap()) {
+  const { readdir } = lazyLoadFsPromises();
+
+  const filenames = await readdir(dir, { withFileTypes: true });
+
+  for await (const file of filenames) {
+    const f = path.join(dir, file.name);
+
+    files.set(f, file);
+
+    if (file.isDirectory()) {
+      await traverse(f, files);
+    }
+  }
+
+  return files;
+}
+
+class FSWatcher extends EventEmitter {
+  #options = null;
+  #closed = false;
+  #files = new SafeMap();
+  #rootPath = path.resolve();
+
+  constructor(options = kEmptyObject) {
+    super();
+
+    validateObject(options, 'options');
+
+    if (options.persistent !== undefined) {
```

Comment:
nit: I think that usually we accept any nullish value as a valid entry to get the default. Optional suggestion:
```suggestion
    if (options.persistent != null) {
```